### PR TITLE
[CLD-8377] Add ability to open invitation modal from a query parameter

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
As part of the Cloud Enteprise Trial redesign, the marketing team is hoping to send an email to a customer containing a link that allows them to land on Mattermost with the invitation modal open. In the past we've had precedent around this for things like opening pricing or purchase modals, but no re-useable way to do so. 

I've added the QueryParamActionController component that's mounted at the root of Mattermost. This component contains a predefined map of action names, corresponding to ModalData, and dispatches openModal for any that match. The component then removes the relevant query parameter from the URL so it's not executed on refresh.

For example, navigating to https://community.mattermost.com/?action=open_invitation_modal will have you land on Mattermost with the invitation modal open. 

As part of this PR I've updated the styling of the No Permissions View for the invitation modal per UX request. 

#### Screenshots

Before:
image

After:
image

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by nickmisasi** [comment]
FYI @emdecr

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/root/root.tsx [hunk 1]
```diff
@@ -486,6 +487,7 @@ export default class Root extends React.PureComponent<Props, State> {
                         <WindowSizeObserver/>
                         <ModalController/>
                         <AnnouncementBarController/>
+                        <QueryParamActionController/>
```
[the remark this answers is not in the document]
Comment (nickmisasi) [reply webapp/channels/src/components/root/root.tsx:490]: In the past when we've had actions to invoke like this they've involved pages in the system console - so I think it makes the most sense to have it at the root layer, that way it'll work in both places

File: webapp/channels/src/components/query_param_actions/query_param_action_controller.test.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {render} from '@testing-library/react';
```
[the remark this answers is not in the document]
Comment (nickmisasi) [reply webapp/channels/src/components/query_param_actions/query_param_action_controller.test.tsx]: Updated!

**Comment by nickmisasi** [comment]
@abhijit-singh @yasserfaraazkhan I'm having some issues with this in the Spinwick - works 100% of the time locally, but won't work on a new tab. 

@yasserfaraazkhan  for now hold off on QA until I figure it out. 

@abhijit-singh, you can have a look at the updated modal by inviting yourself as a guest, then with that guest user, add ?action=open_invitation_modal to the end of the URL, and refresh the page.

File: webapp/channels/src/components/root/root.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickmisasi) [reply webapp/channels/src/components/root/root.tsx]: Discussed this a bit with Harrison while troubleshooting an issue and he explained to me why moving it down would be better, so I've moved to the channel controller.

**Comment by nickmisasi** [comment]
@yasserfaraazkhan you should be all good now

**Review by yasserfaraazkhan — APPROVED** [review]
thank you @nickmisasi 
curious question: If this link is opened in a incognito mode, it'll ask user to log in. After login should the user have this modal opened?

Currently, it doesn't open invite modal. Just lands on channels.

**Review by abhijit-singh — APPROVED** [review]
Looks good @nickmisasi! Just one request - For the no permissions modal, the title `Unable to invite people` should have the font `Metropolis` which is used for headings instead of the current Open Sans.

> If this link is opened in a incognito mode, it'll ask user to log in. After login should the user have this modal opened?

I believe the current behaviour is fine. We can expect them to be logged-in in most cases.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by gabrieljackson — APPROVED** [review]

**Pull request merged**